src/lib/mina_net2/mina_net2.ml

```diff
@@ -1372,726 +121,419 @@ module Pubsub = struct
-                      ~metadata:
-                        [ ("batch_num", `Int batch_num)
-                        ; ("num_batches", `Int num_batches)
-                        ; ( "peers"
-                          , `List (List.map ~f:Peers_no_ids.to_yojson batch) )
-                        ]))) ) ;
-      Deferred.Or_error.return t
-
-let%test_module "coda network tests" =
-  ( module struct
-    let logger = Logger.create ()
-
-    let testmsg =
-      "This is a test. This is a test of the Outdoor Warning System. This is \
-       only a test."
-
-    let pids = Child_processes.Termination.create_pid_table ()
-
-    let setup_two_nodes network_id =
-      let%bind a_tmp = Unix.mkdtemp "p2p_helper_test_a" in
-      let%bind b_tmp = Unix.mkdtemp "p2p_helper_test_b" in
-      let%bind c_tmp = Unix.mkdtemp "p2p_helper_test_c" in
-      let%bind a =
-        create ~all_peers_seen_metric:false
-          ~logger:(Logger.extend logger [ ("name", `String "a") ])
-          ~conf_dir:a_tmp ~pids
-          ~on_unexpected_termination:(fun () ->
-            raise Child_processes.Child_died)
-        >>| Or_error.ok_exn
-      in
-      let%bind b =
-        create ~all_peers_seen_metric:false
-          ~logger:(Logger.extend logger [ ("name", `String "b") ])
-          ~conf_dir:b_tmp ~pids
-          ~on_unexpected_termination:(fun () ->
-            raise Child_processes.Child_died)
-        >>| Or_error.ok_exn
-      in
-      let%bind c =
-        create ~all_peers_seen_metric:false
-          ~logger:(Logger.extend logger [ ("name", `String "c") ])
-          ~conf_dir:c_tmp ~pids
-          ~on_unexpected_termination:(fun () ->
-            raise Child_processes.Child_died)
-        >>| Or_error.ok_exn
-      in
-      let%bind kp_a = Keypair.random a in
-      let%bind kp_b = Keypair.random b in
-      let%bind kp_c = Keypair.random c in
-      let maddrs = [ "/ip4/127.0.0.1/tcp/0" ] in
-      let%bind () =
-        configure a ~logger ~external_maddr:(List.hd_exn maddrs) ~me:kp_a
-          ~maddrs ~network_id ~peer_exchange:true ~mina_peer_exchange:true
-          ~direct_peers:[] ~seed_peers:[] ~on_peer_connected:Fn.ignore
-          ~on_peer_disconnected:Fn.ignore ~flooding:false ~metrics_port:None
-          ~unsafe_no_trust_ip:true ~max_connections:50
-          ~validation_queue_size:150
-          ~initial_gating_config:
-            { trusted_peers = []; banned_peers = []; isolate = false }
-        >>| Or_error.ok_exn
-      in
-      let%bind raw_seed_peers = listening_addrs a >>| Or_error.ok_exn in
-      let seed_peer =
-        Printf.sprintf "%s/p2p/%s"
-          (List.hd_exn raw_seed_peers)
-          (Keypair.to_peer_id kp_a)
+let handle_push_message t push_message =
+  let open Libp2p_ipc.Reader in
+  let open DaemonInterface in
+  let open PushMessage in
+  match push_message with
+  | PeerConnected m ->
+      let peer_id =
+        Libp2p_ipc.unsafe_parse_peer_id (PeerConnected.peer_id_get m)
```

Comment:
Is unsafe necessary?

Reply:
The underlying Network_peer.Peer.Id module has not checks for format of peer ids, and this involves unsafely casting the string representation the libp2p_helper gives us into that type.